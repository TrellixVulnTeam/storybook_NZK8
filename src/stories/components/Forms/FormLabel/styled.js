// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const Label = styled.label``;

export const Wrapper = styled.div`
  ${p =>
    p.inline
      ? 'display: inline-block;'
      : `display: ${p.isClear ? 'inline-flex': 'flex'};
      justify-content: space-between;`}
  color: ${colors.slateGrey};
  font-size: 14px;
  padding-bottom: 4px;
  padding-left: 6px;
`;
