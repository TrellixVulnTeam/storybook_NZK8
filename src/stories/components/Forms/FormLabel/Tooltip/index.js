// @flow

import React from 'react';
import TooltipTrigger from 'react-popper-tooltip';
import { TooltipPlaceholder, TooltipWrapper } from './styled';

/**
 * @param {string[] | string} messages
 * @return {Function}
 */
const getTooltipContent = (messages: string[] | string) => ({
  arrowRef,
  tooltipRef,
  getArrowProps,
  getTooltipProps,
  placement,
}) => {
  return (
    <TooltipWrapper>
      <div
        {...getTooltipProps({
          ref: tooltipRef,
          className: 'tooltip-container',
        })}
      >
        <div
          {...getArrowProps({
            ref: arrowRef,
            className: 'tooltip-arrow',
            'data-placement': placement,
          })}
        />
        {Array.isArray(messages) ? (
          <ul className="tooltip-body">
            {messages.map((message, i) => (
              // eslint-disable-next-line react/no-array-index-key
              <li key={i}>{message}</li>
            ))}
          </ul>
        ) : (
          <div className="tooltip-body tooltip-body__single-message">{messages}</div>
        )}
      </div>
    </TooltipWrapper>
  );
};

type TriggerProps = {
  getTriggerProps: Function,
  triggerRef: React$ElementRef<any>,
};

/**
 * @param {TriggerProps} props
 * @return {React$Node}
 */
const Trigger = ({ getTriggerProps, triggerRef }: TriggerProps) => {
  return (
    <span
      {...getTriggerProps({
        ref: triggerRef,
        className: 'trigger',
      })}
    >
      <TooltipPlaceholder />
    </span>
  );
};

/**
 * @param {string[] | string} messages
 * @return {React$Node}
 */
const Tooltip = ({ messages }: { messages: string[] | string }) => {
  return (
    <TooltipTrigger placement="right-start" trigger="hover" tooltip={getTooltipContent(messages)}>
      {Trigger}
    </TooltipTrigger>
  );
};

export default Tooltip;
