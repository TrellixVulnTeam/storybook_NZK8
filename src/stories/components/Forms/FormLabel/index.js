// @flow

import React from 'react';
import Tooltip from './Tooltip';
import { Wrapper, Label } from './styled';

type Props = {
  children: React$Node,
  htmlFor: string,
  inline?: boolean,
  required?: boolean,
  tooltipMessages?: string[] | string,
  isClear?: boolean,
};

/**
 * Basic form field label that shows above most input fields
 * @example <FormLabel>Email Address</FormLabel>
 * @param {Object} props
 * @return {React$Node}
 */
export default (props: Props) => (
  <Wrapper isClear={props.isClear}>
    <Label htmlFor={props.htmlFor} inline={props.inline} required={props.required}>
      {props.children}
    </Label>
    {props.tooltipMessages && <Tooltip messages={props.tooltipMessages} />}
  </Wrapper>
);
