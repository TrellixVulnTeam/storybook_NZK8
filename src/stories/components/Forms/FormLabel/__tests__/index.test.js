import React from 'react';
import { shallow, mount } from 'enzyme';
import FormLabel from '../index';

describe('<FormLabel />', () => {
  it('snapshot', () => {
    const c = shallow(<FormLabel />);

    expect(c).toMatchSnapshot();
  });

  it('should render props.children', () => {
    const c = mount(<FormLabel>First Name</FormLabel>);

    expect(c.find('label').text()).toContain('First Name');
  });
});
