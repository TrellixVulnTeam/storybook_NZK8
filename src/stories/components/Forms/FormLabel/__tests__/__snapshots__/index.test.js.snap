// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLabel /> snapshot 1`] = `
<styled__Wrapper>
  <styled__Label />
</styled__Wrapper>
`;
